lib/Service/OAuthService.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Stripe\Service;
+
+class OAuthService extends \Stripe\Service\AbstractService
+{
+    /**
+     * Generates a URL to Stripe's OAuth form.
+     *
+     * @param null|array $params
+     * @param null|array $opts
+     *
+     * @return string the URL to Stripe's OAuth form
+     */
+    public function authorizeUrl($params = null, $opts = null)
+    {
+        $params = $params ?: [];
+
+        $base = ($opts && \array_key_exists('connect_base', $opts)) ? $opts['connect_base'] : $this->client->getConnectBase();
+
+        $clientId = $this->_getClientId($params);
+        $params['client_id'] = $this->_getClientId($params);
+        if (!\array_key_exists('response_type', $params)) {
+            $params['response_type'] = 'code';
+        }
+        $query = \Stripe\Util\Util::encodeParameters($params);
+
+        return $base . '/oauth/authorize?' . $query;
+    }
+
+    /**
+     * Use an authoriztion code to connect an account to your platform and
+     * fetch the user's credentials.
+     *
+     * @param null|array $params
+     * @param null|array $opts
+     *
+     * @throws \Stripe\Exception\OAuth\OAuthErrorException if the request fails
+     *
+     * @return StripeObject object containing the response from the API
+     */
+    public function token($params = null, $opts = null)
+    {
+        return $this->requestConnect('post', $this->buildPath('/oauth/token'), $params, $opts);
```

Comment:
We should fill the `client_secret` parameter with the API key if it's not explicitly provided.

```suggestion
        $params = $params ?: [];
        if (!\array_key_exists('client_secret', $params)) {
            $params['client_secret'] = $this->client->getApiKey();
        }
        return $this->requestConnect('post', $this->buildPath('/oauth/token'), $params, $opts);
```